### PR TITLE
fix issue 1770:

### PR DESCRIPTION
**Pull request by zg009**
changed error handling of attempting to send password reset on nonexistent user: view issue 1770

@bourgeoa , I wasn't sure what you meant by changing that one line, as I could not find a fix without changing how the accountManager worked downstream, so I simply added an error code check to if request.validate() fails it still passes through the catch.

If you have a better idea let me know and I will change + test

### CODE REVIEW

File: lib/requests/password-reset-email-request.js [hunk 1]
```diff
@@ -123,7 +130,9 @@ class PasswordResetEmailRequest extends AuthRequest {
     return this.accountManager.accountExists(username)
       .then(exists => {
         if (!exists) {
-          throw new Error('Account not found for that username')
+          const error = new Error("Account not found for that username")
+          error.code = "ACCOUNT_MISSING"
+          throw error
```
Comment (bourgeoa) [inline lib/requests/password-reset-email-request.js:135]:
```suggestion
          // For security avoid using an error that leak information
          return renderSuccess()
```
`renderSuccess()` is may be changed to better name like resetLinkMessage()
The message itself should be changed in 
https://github.com/nodeSolidServer/node-solid-server/blob/d8290cbe3f7fe5ba7ccd46e988a1591659c6de93/default-views/auth/reset-link-sent.hbs#L17

File: lib/requests/password-reset-email-request.js [hunk 2]
```diff
@@ -93,6 +93,13 @@ class PasswordResetEmailRequest extends AuthRequest {
     return Promise.resolve()
       .then(() => request.validate())
       .then(() => request.loadUser())
+      .catch((err) => {
+        if (err.code === "ACCOUNT_MISSING") {
+          this.response.render('auth/reset-link-sent')
+        } else {
+          return err;
+        }
+      })
```
Comment (bourgeoa) [inline lib/requests/password-reset-email-request.js:102]:
```suggestion
```

**Review by bourgeoa — CHANGES_REQUESTED** [review]

**Comment by bourgeoa** [comment]
@zg009 
Do you work locally ? if yes you should  `npm run test` before push. If you need any help please ask.

Anyway after a PR it is good ppractice to come back and look a the tests. Thereafter clear any issue due to your changes.

**Comment by zg009** [comment]
> @zg009 I'm sorry I have not been very clear to the changes I was thinking of.
> 
> Do you work locally ? if yes you should `npm run test` before push. If you need any help please ask.
> 
> Usually after a PR it is good practice to come back and look a the tests. Thereafter clear any issue due to your changes.

I saw, I thought I had passed the linting check but had forgotten to save, so I committed without fixing the formatting issues.

File: lib/requests/password-reset-email-request.js (hunk 1 above)
Comment (zg009) [reply lib/requests/password-reset-email-request.js]: I changed the wording and renamed it to resetLinkMessage in the latest commits

File: lib/requests/password-reset-email-request.js (hunk 2 above)
Comment (zg009) [reply lib/requests/password-reset-email-request.js]: I removed this as well

**Pull request closed without merging**